libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -361,24 +361,22 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 
 void DeclarationTypeChecker::typeError(SourceLocation const& _location, string const& _description)
 {
-	m_errorOccurred = true;
 	m_errorReporter.typeError(2311_error, _location, _description);
```

Comment:
I guess this will still be fixed, right?

Reply:
Yes. The idea is that `DeclarationTypeChecker::typeError()`) will disappear and a regular `m_errorReporter.typeError(3145_error, ....)` will be used instead.

I hope this is OK, unless we want `DeclarationTypeChecker::check()` to return `false` for some specific errors and `true` for others.

Reply:
Yes, sounds great!